### ISSUE TITLE
tf.pow gives some incorrect results on cuda: tf.pow(2.0,13)==8191.996 

### ISSUE THREAD

**Issue by gilfree** [tensorflow/tensorflow#56998]
### Issue Type

Bug

### Source

binary

### Tensorflow Version

tf 2.8

### Custom Code

Yes

### OS Platform and Distribution

ubuntu 18.04

### Python version

3.7

### CUDA/cuDNN version

11.4

### GPU model and memory

RTX 2080TI 11GB RAM

### Current Behaviour?

```shell
On Cuda only (Rtx 2080TI, Cuda 11.4, and also other gpus, driver 515.48.07)

The result of tf.pow is incorrect to some numbers, for example:

`print(tf.pow(2.0,13))`

prints:

`tf.Tensor(8191.996, shape=(), dtype=float32)`

2.0**13 is well within float32 precision and is 8192.

This *does not* happen on TF 2.2
```

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
print(tf.pow(2.0,13))
```

**Comment by sushreebarsa** [comment]
@gadagashwini I was able to replicate the issue on colab, please find the [gist](https://colab.research.google.com/gist/sushreebarsa/ac10ff2133c46bb46aa0943d4490d436/56998.ipynb) here. Thank you!

**Comment by gadagashwini** [comment]
Hi @gilfree, Below scenarios tested with Tensorflow 2.9.1
I tried with CUDA 11.4 on Ubuntu. I could able to get `tf.pow(2.0,13)==8191.996`.
Where as in Colab, result slightly differ `tf.pow(2.0,13)==8192.0`.
Could you confirm issue. Thank you!

**Comment by gilfree** [comment]
Hi. 
I don't have collab access currently, but can confirm that the bug happens on TF 2.9.1 with python 3.7 and cuda 11.7 on ubuntu.
Any other check you want need?

**Comment by gadagashwini** [comment]
I could able to reproduce the issue with CUDA 11.2 and CUDA 11.4, `tf.pow(2.0,13)==8191.996`. Thank you!

**Comment by sachinprasadhs** [comment]
@gilfree , I think the issue is specific to GPU, I was able to reproduce the issue on colab with GPU runtime.
But, when the inputs are changed to float64 precision, the results are as expected. 
[Here](https://colab.sandbox.google.com/gist/sachinprasadhs/200530442d713455b7bfa3bf1958a0d4/56998.ipynb) is the attached gist for reference.

**Comment by gilfree** [comment]
As I have stated in the bug description this is indeed CUDA only issue. 

Any kind of floating point number that is in use today should easily be able to represent this computation - be it float16, bfloat16, float32, nvidia's tensor float (see here: https://en.wikipedia.org/wiki/Bfloat16_floating-point_format and here: https://oletus.github.io/float16-simulator.js/ for simulating this computation).

For example, compare, on the same machine:

tensorflow 2.8 (happens form 2.5 onwards):

```
>>> import tensorflow
>>> tensorflow.__version__
'2.8.0'
>>> tensorflow.pow(2.0,13)
<tf.Tensor: shape=(), dtype=float32, numpy=8191.996>
>>> tensorflow.pow(2.0,13).device
'/job:localhost/replica:0/task:0/device:GPU:0'
>>> tf.pow(tf.cast(2.0,dtype='float16'),13)
<tf.Tensor: shape=(), dtype=float16, numpy=8190.0>
>>> tf.pow(tf.cast(2.0,dtype='float16'),13).device
'/job:localhost/replica:0/task:0/device:GPU:0'
```

tensorflow 2.2
```
>>> tf.pow(2.0,13)
<tf.Tensor: shape=(), dtype=float32, numpy=8192.0>
>>> tf.__version__
'2.2.3'
```

torch:

```
>>> import torch
>>> torch.pow(torch.as_tensor(2.0,dtype=torch.float32,device='cuda'),13)
tensor(8192., device='cuda:0')
>>> torch.__version__
'1.11.0+cu113'
```

cupy:

```
>>> import cupy
>>> cupy.__version__
'11.0.0'
>>> cupy.power(cupy.asarray(2.0,dtype='float32'),13)
array(8192., dtype=float32)
>>> cupy.power(cupy.asarray(2.0,dtype='float32'),13).device
<CUDA Device 0>
>>> cupy.power(cupy.asarray(2.0,dtype='float16'),13)
array(8192., dtype=float16)
>>> cupy.power(cupy.asarray(2.0,dtype='float16'),13).device
<CUDA Device 0>
```

For float64, the error in tensorflow starts at a larger number but still worse then other libs:
```
<tf.Tensor: shape=(), dtype=float64, numpy=33554432.0>
>>> tf.pow(tf.cast(2.0,'float64'),26)
<tf.Tensor: shape=(), dtype=float64, numpy=67108863.99999999>
>>> cupy.power( cupy.asarray(2.0,dtype='float64'),26)
array(67108864.)
```

(The installed cuda version used by cupy in this case is cuda 11.4, the same one used by TF)


This kind of precision loss at this range is really something that should not happen, and any numerical library (that I have tested) but tensorflow recent versions, on any processor, does this correctly - TF 2.2 also worked correctly.

This is clearly some regression with new tensorflow versions.

To summarize when this bug happens:
- On CUDA. At all tested cuda 11 sub-versions. 
- Only on TF>=2.5. 
- All tested GPU architectures (Turing, Ampre).
- In FP32, and FP16 and in FP64, for other numbers.
- It does not happen in other libraries like pytorch or cupy.

**Comment by joshua-xia** [comment]
I reproducted this issue on tf 2.9.1

**Comment by joshua-xia** [comment]
tf.pow(tf.cast(2.0,dtype=tf.float16,13)
<tf.Tensor: shape=(), dtype=float16, numpy=8192.0>

tf.pow(tf.cast(2.0,dtype=tf.float32,13)
<tf.Tensor: shape=(), dtype=float32, numpy=8191.996>

tf.pow(tf.cast(2.0,dtype=tf.float64,13)
<tf.Tensor: shape=(), dtype=float64, numpy=8192.0>

**Comment by mohantym** [comment]
@gilfree @joshua-xia !
I am able to replicate this issue in 2.11 , Attached [gist](https://colab.sandbox.google.com/gist/mohantym/d170b04b0908b2d4bff3a51d35c30c53/56998.ipynb#scrollTo=qcT55Y7O_IY-) for reference.
Thank you!
